free_fleet_adapter/free_fleet_adapter/nav1_robot_adapter.py

```diff
@@ -0,0 +1,553 @@
+                f'Navigation goal {nav_goal_id} is not found, replan count '
+                f'[{self.replan_counts}]'
+            )
+            self.update_handle.more().replan()
+            return False
+
+        match goal_status.status:
+            case goal_status.PENDING | \
+                 goal_status.ACTIVE | \
+                 goal_status.PREEMPTING | \
+                 goal_status.RECALLING:
+                return False
+            case goal_status.SUCCEEDED:
+                self.node.get_logger().info(
+                    f'Navigation goal {nav_goal_id} reached'
+                )
+                return True
+            case goal_status.PREEMPTED:
+                self.node.get_logger().info(
+                    f'Navigation goal {nav_goal_id} was cancelled'
+                )
+                return True
+            case _:
+                # TODO(ac): test replanning behavior if goal status is
+                # neither executing or succeeded
+                self.replan_counts += 1
+                self.node.get_logger().error(
+                    f'Navigation goal {nav_goal_id} status '
+                    f'{goal_status.status}, replan count '
+                    f'[{self.replan_counts}]'
+                )
+                self.update_handle.more().replan()
+                return False
+
+    def _check_update_handle_initialization(self):
+        if self.update_handle is None:
+            error_message = \
+                f'Failed to update robot {self.name}, robot adapter has ' \
+                'not yet been initialized with a fleet update handle.'
+            self.node.get_logger().error(error_message)
+            raise RuntimeError(error_message)
+
+    def update(self, state: rmf_easy.RobotState):
+        self._check_update_handle_initialization()
+
+        activity_identifier = None
+        if self.execution:
+            if self.nav_goal_id is not None and self._is_navigation_done():
+                if self.execution is not None:
+                    self.execution.finished()
+                    self.execution = None
+                self.nav_goal_id = None
+                self.replan_counts = 0
+            else:
+                activity_identifier = self.execution.identifier
+
+        self.update_handle.update(state, activity_identifier)
+
+    def _handle_navigate_to_pose(
+        self,
+        map_name: str,
+        x: float,
+        y: float,
+        z: float,
+        yaw: float,
+        timeout_sec: float = 3.0
+    ):
+        if map_name != self.map_name:
+            # TODO(ac): test this map related replanning behavior
+            self.replan_counts += 1
+            self.node.get_logger().error(
+                f'Destination is on map [{map_name}], while robot '
+                f'[{self.name}] is on map [{self.map_name}], replan count '
+                f'[{self.replan_counts}]'
+            )
+
+            self._check_update_handle_initialization()
```

Comment:
Didn't catch this the previous time, but this check is not required here. If RMF is able to call `navigate()`, it means an [update handle is already successfully created](https://github.com/open-rmf/free_fleet/blob/easy-full-control/free_fleet_adapter/free_fleet_adapter/fleet_adapter.py#L233-L247). Same for nav2 adapter.

Reply:
removed in https://github.com/open-rmf/free_fleet/pull/145/commits/cccb4033c024b5035a97c7cda146f70970b24baf